File: lib/core/models/operation_result.dart
/// معادل OperationResult.cs
class OperationResult {
  final bool success;
  final String? errorMessage;
  final String? filePath;

  OperationResult._({required this.success, this.errorMessage, this.filePath});

  factory OperationResult.ok(String filePath) =>
      OperationResult._(success: true, filePath: filePath);

  factory OperationResult.fail(String error) =>
      OperationResult._(success: false, errorMessage: error);
}

File: windows/SFE.Core/Models/OperationResult.cs
namespace SFE.Core.Models;

public class OperationResult
{
    public bool Success { get; set; }
    public string? ErrorMessage { get; set; }
    public string? FilePath { get; set; }

    public static OperationResult Ok(string filePath) =>
        new() { Success = true, FilePath = filePath };

    public static OperationResult Fail(string error) =>
        new() { Success = false, ErrorMessage = error };
}